source/code/troubleshooter/modules/errors.py

```diff
@@ -0,0 +1,402 @@
+          "Please set the NSS_SDB_USE_CACHE environment variable to 'yes'. You can check the "\
+          "below link for more information:\n"\
+          "\n    https://docs.microsoft.com/en-us/azure/azure-monitor/platform/agent-linux-troubleshoot#issue-you-see-omiagent-using-100-cpu",
+    ERR_LOGROTATE_SIZE : "Logrotate size limit for log {0} has invalid formatting. Please see {1} for more "\
+          "information.",
+    ERR_LOGROTATE : "Logrotate isn't rotating log {0}: its current size is {1}, and it should have "\
+          "been rotated at {2}. Please see {3} for more information.",
+    WARN_LARGE_FILES : "File {0} has been modified {1} times in the last {2} seconds.",
+    ERR_PKG : "{0} isn't installed correctly.",
+    ERR_BACKEND_CONFIG : "The agent is currently having issues with pulling the configuration from the backend. "\
+          "You can try manually pulling the config by running the below command:\n"\
+          "\n  $ sudo su omsagent -c 'python /opt/microsoft/omsconfig/Scripts/PerformRequiredConfigurationChecks.py'\n\n"\
+          "Please check the following log files for more information:\n"\
+          "    {0}\n"\
+          "    {1}\n"\
+          "Please also check out the below link for more information:\n"\
+          "\n    https://docs.microsoft.com/en-us/azure/azure-monitor/platform/agent-linux-troubleshoot#issue-you-are-not-seeing-any-custom-log-data",
+    ERR_PYTHON_PKG : "This version of Python is missing the {0} package. (You can check by opening up "\
+          "python and typing 'import {0}'.) Please install this package and run the troubleshooter again."
+
+}
+
+
+
+# check if either has no error or is warning
+def is_error(err_code):
+    not_errs = warnings.copy()
+    not_errs.add(NO_ERROR)
+    return (err_code not in not_errs)
+
+
+
+# TODO: grab these directly from https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/oms-linux#error-codes-and-their-meanings
+extension_err_codes_dict = {
+    '9'  : "Enable called prematurely, try updating the Azure Linux Agent to the latest available version",
+    '10' : "VM is already connected to a Log Analytics workspace",
+    '11' : "Invalid config provided to the extension",
+    '17' : "Log Analytics package installation failure",
+    '19' : "OMI package installation failure",
+    '20' : "SCX package installation failure",
+    '51' : "This extension is not supported on the VM's operation system",
+    '55' : "Cannot connect to the Azure Monitor service, or required packages missing, or dpkg package manager is locked"
+}
+
+# get error codes from Troubleshooting.md
+def get_error_codes(err_type):
+    # extension
+    if (err_type == 'Extension'):
+        return (extension_err_codes_dict, None)
+
+    # shell bundle or onboarding
+    err_codes = {0 : "No errors found"}
+    section_name = "{0} Error Codes".format(err_type)
+    try:
+        ts_doc = urlopen(TSG_URL)
```

Comment:
Change error of can't connect to Github to warning (bc not necessary for rest of troubleshooter)

Reply:
Fixed here:
https://github.com/microsoft/OMS-Agent-for-Linux/blob/bff7c80f869d6f5a4046ed41b4bd0ed89d69219f/source/code/troubleshooter/modules/errors_tsg.py#L58-L64